styli.sh

```diff
@@ -121,109 +128,106 @@ colors:
-useragent="thevinter"
-timeout=60
+reddit() {
+    USERAGENT="thevinter"
+    TIMEOUT=60
 
-sort=$2
-top_time=$3
-if [ -z $sort ]; then
-    sort="hot"
-fi
+    SORT="$2"
+    TOP_TIME="$3"
+    if [ -z "$SORT" ]; then
+        SORT="hot"
+    fi
 
-if [ -z $top_time ]; then
-    top_time=""
-fi
+    if [ -z "$TOP_TIME" ]; then
+        TOP_TIME=""
+    fi
 
-if [ ! -z $1 ]; then
-    sub=$1
-else
-    if [ ! -f "${confdir}/subreddits" ]; then
-        echo "Please install the subreddits file in ${confdir}"
-        exit 2
+    if [ -n "$1" ]; then
+        SUB="$1"
+    else
+        if [ ! -f "$CONFDIR/subreddits" ]; then
+            echo "Please install the subreddits file in $CONFDIR"
+            exit 2
+        fi
+        mapfile -t SUBREDDITS <<<"$CONFDIR/subreddits"
+        a=${#SUBREDDITS[@]}
+        b=$((RANDOM % a))
+        SUB=${SUBREDDITS[$b]}
+        SUB="$(echo -e "$SUB" | tr -d '[:space:]')"
     fi
-    readarray subreddits < "${confdir}/subreddits"
-    a=${#subreddits[@]}
-    b=$(($RANDOM % $a))
-    sub=${subreddits[$b]}
-    sub="$(echo -e "${sub}" | tr -d '[:space:]')"
-fi
 
-url="https://www.reddit.com/r/$sub/$sort/.json?raw_json=1&t=$top_time"
-content=`wget -T $timeout -U "$useragent" -q -O - $url`
-urls=$(echo -n "$content"| jq -r '.data.children[]|select(.data.post_hint|test("image")?) | .data.preview.images[0].source.url')
-names=$(echo -n "$content"| jq -r '.data.children[]|select(.data.post_hint|test("image")?) | .data.title')
-ids=$(echo -n "$content"| jq -r '.data.children[]|select(.data.post_hint|test("image")?) | .data.id')
-arrURLS=($urls)
-arrNAMES=($names)
-arrIDS=($ids)
-wait # prevent spawning too many processes
-size=${#arrURLS[@]}
-if [ $size -eq 0 ]; then
-    echo The current subreddit is not valid.
-    exit 1
-fi
-idx=$(($RANDOM % $size))
-target_url=${arrURLS[$idx]}
-target_name=${arrNAMES[$idx]}
-target_id=${arrIDS[$idx]}
-ext=`echo -n "${target_url##*.}"|cut -d '?' -f 1`
-newname=`echo $target_name | sed "s/^\///;s/\// /g"`_"$subreddit"_$target_id.$ext
-wget -T $timeout -U "$useragent" --no-check-certificate -q -P down -O ${wallpaper} $target_url &>/dev/null
+    URL="https://www.reddit.com/r/$SUB/$SORT/.json?raw_json=1&t=$TOP_TIME"
+    CONTENT=$(wget -T $TIMEOUT -U "$USERAGENT" -q -O - "$URL")
+    mapfile -t URLS <<<"$(echo -n "$CONTENT" | jq -r '.data.children[]|select(.data.post_hint|test("image")?) | .data.preview.images[0].source.url')"
+    wait # prevent spawning too many processes
+    SIZE=${#URLS[@]}
+    if [ "$SIZE" -eq 0 ]; then
+        echo The current subreddit is not valid.
+        exit 1
+    fi
+    IDX=$((RANDOM % SIZE))
+    # TARGET_NAME, TARGET_ID, EXT and NEWNAME are not used
+    # mapfile -t IDS <<<"$(echo -n "$CONTENT" | jq -r '.data.children[]|select(.data.post_hint|test("image")?) | .data.id')"
+    # mapfile -t NAMES <<<"$(echo -n "$CONTENT" | jq -r '.data.children[]|select(.data.post_hint|test("image")?) | .data.title')"
+    # TARGET_URL=${URLS[$IDX]}
```

Comment:
This line should be uncommented, am I missing something?

Reply:
Hey, my mistake. It shouldn't.